### PR TITLE
Add breadcrumbs to product reviews

### PR DESCRIPTION
**Pull request by bartek-sek**
image

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/Twig/Component/Common/ProductSlugComponent.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public function __construct(
+        private readonly ProductRepositoryInterface $productRepository,
+        private readonly ChannelContextInterface $channelContext,
+        private readonly LocaleContextInterface $localeContext,
+    ) {
+    }
+
+    #[ExposeInTemplate(name: 'product')]
+    public function getProductBySlug(): ProductInterface
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ShopBundle/Twig/Component/Common/ProductSlugComponent.php]:
```suggestion
    public function getProductBySlug(): ?ProductInterface
```

**Review by Wojdylak — DISMISSED** [review]

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml [hunk 2]
```diff
@@ -41,5 +41,16 @@
 
             <tag name="sylius.twig_component" key="sylius_shop:common:taxon_menu" />
         </service>
+
+        <service
+                id="sylius_shop.twig.component.common.product_slug"
+                class="Sylius\Bundle\ShopBundle\Twig\Component\Common\ProductSlugComponent"
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml]:
```suggestion
                class="Sylius\Bundle\ShopBundle\Twig\Component\Product\BySlugComponent"
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml [hunk 3]
```diff
@@ -41,5 +41,16 @@
...
+        >
+            <argument type="service" id="sylius.repository.product" />
+            <argument type="service" id="sylius.context.channel" />
+            <argument type="service" id="sylius.context.locale" />
+
+            <tag name="sylius.twig_component" key="sylius_shop:common.product_slug" />
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml]: And according to the above suggestion:
```suggestion
            <tag name="sylius.twig_component" key="sylius_shop:product:by_slug" />
```

File: src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{% macro product_review_breadcrumbs(product, new_items) %}
+    {% from '@SyliusShop/shared/breadcrumbs.html.twig' import breadcrumbs as breadcrumbs %}
+
+    {% set items = [] %}
+
+    {% set items = items|merge([{ path: path('sylius_shop_homepage'), label: 'sylius.ui.home'|trans }]) %}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig]:
```suggestion
    {% set items = items|default([])|merge([{ path: path('sylius_shop_homepage'), label: 'sylius.ui.home'|trans }]) %}
```

File: src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig:1]: Couldn't we use the same template that is used for product show page? 🤔

File: src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    {% if product.mainTaxon is not null %}
+        {% set taxon = product.mainTaxon %}
+        {% set ancestors = taxon.ancestors|reverse %}
+
+        {% for ancestor in ancestors %}
+            {% if ancestor.isRoot()  or not ancestor.enabled %}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig]:
```suggestion
            {% if ancestor.isRoot() or not ancestor.enabled %}
```

File: src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig
Comment (bartek-sek) [reply src/Sylius/Bundle/ShopBundle/templates/product_review/common/macro/breadcrumbs.html.twig:1]: Yes we could. Done :)

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Bundle/ShopBundle/templates/product_review/create/header/breadcrumbs.html.twig [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{% from '@SyliusShop/product/common/macro/breadcrumbs.html.twig' import product_breadcrumbs as product_breadcrumbs %}
+
+{% set additioinal_items = [
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/product_review/create/header/breadcrumbs.html.twig]:
```suggestion
{% set additional_items = [
```

File: src/Sylius/Bundle/ShopBundle/templates/product_review/create/header/breadcrumbs.html.twig [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    { label: 'sylius.ui.reviews'|trans, path: path('sylius_shop_product_review_index', {'slug': product.slug, '_locale': product.translation.locale}) },
+    { label: 'sylius.ui.add'|trans, active: true },
+] %}
+
+{{ product_breadcrumbs(product, additioinal_items) }}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/product_review/create/header/breadcrumbs.html.twig]:
```suggestion
{{ product_breadcrumbs(product, additional_items) }}
```

**Review by Rafikooo — APPROVED** [review]

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml [hunk 8]
```diff
@@ -41,5 +41,16 @@
 
             <tag name="sylius.twig_component" key="sylius_shop:common:taxon_menu" />
         </service>
+
+        <service
+                id="sylius_shop.twig.component.product.by_slug"
+                class="Sylius\Bundle\ShopBundle\Twig\Component\Product\BySlugComponent"
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml:47]:
```suggestion
            id="sylius_shop.twig.component.product.by_slug"
            class="Sylius\Bundle\ShopBundle\Twig\Component\Product\BySlugComponent"
```

**Review by GSadee — APPROVED** [review]

**Pull request merged**